cmd/influxd/restore/restore.go

```diff
@@ -10,138 +11,126 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/BurntSushi/toml"
+	"github.com/influxdb/influxdb/cmd/influxd/backup"
 	"github.com/influxdb/influxdb/meta"
-	"github.com/influxdb/influxdb/snapshot"
-	"github.com/influxdb/influxdb/tsdb"
 )
 
 // Command represents the program execution for "influxd restore".
 type Command struct {
 	Stdout io.Writer
 	Stderr io.Writer
+
+	backupFilesPath string
+	metadir         string
+	datadir         string
+	database        string
+	retention       string
+	shard           string
+
+	// TODO: when the new meta stuff is done this should not be exported or be gone
+	MetaConfig *meta.Config
 }
 
 // NewCommand returns a new instance of Command with default settings.
 func NewCommand() *Command {
 	return &Command{
-		Stdout: os.Stdout,
-		Stderr: os.Stderr,
+		Stdout:     os.Stdout,
+		Stderr:     os.Stderr,
+		MetaConfig: meta.NewConfig(),
 	}
 }
 
 // Run executes the program.
 func (cmd *Command) Run(args ...string) error {
-	config, path, err := cmd.parseFlags(args)
-	if err != nil {
+	if err := cmd.parseFlags(args); err != nil {
 		return err
 	}
 
-	return cmd.Restore(config, path)
-}
-
-// Restore restores a database snapshot
-func (cmd *Command) Restore(config *Config, path string) error {
-	// Remove meta and data directories.
-	if err := os.RemoveAll(config.Meta.Dir); err != nil {
-		return fmt.Errorf("remove meta dir: %s", err)
-	} else if err := os.RemoveAll(config.Data.Dir); err != nil {
-		return fmt.Errorf("remove data dir: %s", err)
-	}
-
-	// Open snapshot file and all incremental backups.
-	mr, files, err := snapshot.OpenFileMultiReader(path)
-	if err != nil {
-		return fmt.Errorf("open multireader: %s", err)
+	if cmd.metadir != "" {
+		if err := cmd.unpackMeta(); err != nil {
+			return err
+		}
 	}
-	defer closeAll(files)
 
-	// Unpack files from archive.
-	if err := cmd.unpack(mr, config); err != nil {
-		return fmt.Errorf("unpack: %s", err)
+	if cmd.shard != "" {
+		return cmd.unpackShard(cmd.shard)
+	} else if cmd.retention != "" {
+		return cmd.unpackRetention()
 	}
 
-	// Notify user of completion.
-	fmt.Fprintf(os.Stdout, "restore complete using %s", path)
-	return nil
+	return cmd.unpackDatabase()
 }
 
 // parseFlags parses and validates the command line arguments.
-func (cmd *Command) parseFlags(args []string) (*Config, string, error) {
+func (cmd *Command) parseFlags(args []string) error {
 	fs := flag.NewFlagSet("", flag.ContinueOnError)
-	configPath := fs.String("config", "", "")
+	fs.StringVar(&cmd.metadir, "metadir", "", "")
```

Comment:
the `Usage` function is set to something that returns everything. See the `printUsage` function.